src/rust/catnap/mod.rs

```diff
@@ -301,7 +299,7 @@ impl CatnapLibOS {
         match queue.do_close(yielder).await {
             Ok(()) => {
                 // Remove the queue from the queue table.
-                runtime.free_queue::<CatnapQueue>(qd);
+                runtime.free_queue::<CatnapQueue>(&qd).expect("queue should exist");
```

Comment:
To follow out guidelines, could you add a comment here to explain why it is safe to call `expect()` here?